core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -2523,6 +2523,16 @@ public static int subtractMonths(long t0, long t1) {
     return x;
   }
 
+
+  public static Integer dateStringToUnixDate(String s) {
+    return s == null ? null : DateTimeUtils.dateStringToUnixDate(s);
```

Comment:
Accept null args is another topic, i think use the `DateTimeUtils.dateStringToUnixDate` directly is ok for this patch.

Reply:
OK